src/aws_credentials_ec2.erl

```diff
@@ -28,7 +28,7 @@ fetch_role() ->
     {ok, Body}.
 
 fetch_metadata(Role) ->
-    Url = lists:flatten(io_lib:format("~p/~p", [?CREDENTIAL_URL, Role])),
+    Url = lists:flatten(io_lib:format("~s~s", [?CREDENTIAL_URL, Role])),
```

Comment:
Should we consider adding a test for this?

Reply:
Indeed. The whole application requires some love when it comes to testing. I will contribute some as part of a separate PR.

Reply:
Awesome!